### PR TITLE
SD-1255 Display correct currency for customer return's pre-tax total

### LINKED ISSUE #10925: Multi Currency & Admin Panel: Customer Returns - Pre-tax total is displayed in incorrect currency
## Context
It has been found that customer return, in order, is displayed with USD currency, not in the currency which in order was placed and paid (GBP or Euro). The currency displayed is the one that is set in store configuration.
Order on the screenshot was placed in GBP currency.

image (4)

## Expected Behavior
Pre-Tax Total is displayed in the currency the order was placed in

## Actual Behavior
Pre-Tax Total is displayed in USD.

## Steps to Reproduce
1. Place an order in other than USD (use GBP or Euro)
2. Go to AP and process the order to shipment
3. Create a Return Authorization
4. Create a Customer Return
5. Observe Pre-Tax Total on this Customer Return

## Your Environment
Affected version: spee.4.3.0.alpha (ruby 2.7.2; rails 6.0.0)
Configuration:
gem ‘spree’
gem ‘spree_gateway’
gem ‘spree_auth_devise’
gem ‘spree_static_content’
gem ‘spree_related_products’
gem ‘spree_sitemap’
gem ‘spree_analytics_trackers’
gem ‘spree_braintree_vzero’
gem ‘spree_i18n’

### PR DESCRIPTION
**Pull request by tomurb**
Fixes https://github.com/spree/spree/issues/10925

### CODE REVIEW

File: core/app/models/spree/customer_return.rb [hunk 1]
```diff
@@ -16,9 +16,6 @@ class CustomerReturn < Spree::Base
 
     accepts_nested_attributes_for :return_items
 
-    extend DisplayMoney
-    money_methods pre_tax_total: { currency: Spree::Config[:currency] }
```
Comment (damianlegawiec) [inline core/app/models/spree/customer_return.rb:20]:
```ruby
money_methods :pre_tax_total
```

plus 

```ruby
def currency
  order.currency || Store.default.default_currency
end
```

Comment (tomurb) [reply core/app/models/spree/customer_return.rb:20]: Maybe I'll just delegate `currency` without ` || Store.default.default_currency`?
```
# core/app/models/spree/order.rb:723
    def ensure_currency_presence
      self.currency ||= store.default_currency || Spree::Config[:currency]
    end
```

Comment (damianlegawiec) [reply core/app/models/spree/customer_return.rb:20]: @tomurb that will also do

File: core/app/models/spree/customer_return.rb [hunk 2]
```diff
@@ -31,6 +31,10 @@ def fully_reimbursed?
       completely_decided? && return_items.accepted.includes(:reimbursement).all? { |return_item| return_item.reimbursement.try(:reimbursed?) }
     end
 
+    def currency
+      order.currency
```
Comment (damianlegawiec) [inline core/app/models/spree/customer_return.rb:35]:
```suggestion
      order&.currency
```

order can be nil in some cases

File: core/app/models/spree/customer_return.rb (hunk 1 above)
Comment (damianlegawiec) [reply core/app/models/spree/customer_return.rb:20]: @tomurb please also add a test for this feature :)

**Review by damianlegawiec — APPROVED** [review]

File: core/spec/models/spree/customer_return_spec.rb [hunk 3]
```diff
@@ -83,6 +83,19 @@
...
+    subject { customer_return.display_pre_tax_total }
+
+    let(:pre_tax_amount)  { 15.0 }
+    let(:customer_return) { create(:customer_return, line_items_count: 2) }
+
+    before do
+      Spree::ReturnItem.where(customer_return_id: customer_return.id).update_all(pre_tax_amount: pre_tax_amount)
+    end
+
+    it { is_expected.to eq Spree::Money.new(pre_tax_amount * 2) }
```
Comment (damianlegawiec) [inline core/spec/models/spree/customer_return_spec.rb:96]:
```suggestion
    it { is_expected.to eq '$ 30' }
```

we need to test the final output

Comment (damianlegawiec) [reply core/spec/models/spree/customer_return_spec.rb:96]: @tomurb also let's add a test for `currency` method as well

File: core/spec/models/spree/customer_return_spec.rb [hunk 4]
```diff
@@ -107,6 +120,19 @@
     end
   end
 
+  describe '#currency' do
+    subject { customer_return.currency }
+
+    let(:return_item) { create(:return_item) }
+    let(:customer_return) { build(:customer_return, return_items: [return_item]) }
+    let(:order) { create(:order, currency: test_currency) }
+    let(:test_currency) { 'TEST' }
```
Comment (damianlegawiec) [inline core/spec/models/spree/customer_return_spec.rb:129]:
```suggestion
    let(:test_currency) { 'USD' }
```

let's operate on real currencies as in the future we will likely add some validation here

**Pull request merged**